### ISSUE TITLE
Stripe clean up user bug

### ISSUE THREAD

**Issue by SamMatthewsIsACommonName** [firebase/functions-samples#126]
The clean up user function is a little at odds with the stripe API and returned an error for me. It should be changed to:

````
// When a user deletes there account, clean up after them
exports.cleanupUser = functions.auth.user().onDelete(event => {
  return admin.database().ref(`/stripe_customers/${event.data.uid}`).once('value').then(snapshot => {
    return snapshot.val();
  }).then(customer => {
// change below, customer.customer_id vs customer
    return stripe.customers.del(customer.customer_id);
  }).then(() => {
    return admin.database().ref(`/stripe_customers/${event.data.uid}`).remove();
  });
});
````
OR change the reference to admin.database().ref(`/stripe_customers/${event.data.uid}/customer_id`)
Thanks

**Comment by katowulf** [comment]
@SamMatthewsIsACommonName can you please follow the issue template and include the necessary details here? Cheers.

**Comment by SamMatthewsIsACommonName** [comment]
Well I don't think it really needs to be replicated much as it seems to be just one of couple of typos (the other one I saw is int he database rules $sourceId and $chargeId are around the wrong way).
I think the author intended to follow the same pattern as when adding a payment source, by getting a read from 

admin.database().ref(`/stripe_customers/${event.params.userId}/customer_id`).once('value')

to get the stripe customer id as a string:

// Add a payment source (card) for a user by writing a stripe payment source token to Realtime database
exports.addPaymentSource = functions.database.ref('/stripe_customers/{userId}/sources/{pushId}/token').onWrite(event => {
  const source = event.data.val();
  if (source === null) return null;
  return admin.database().ref(`/stripe_customers/${event.params.userId}/customer_id`).once('value').then(snapshot => {
    return snapshot.val();
  }).then(customer => {
    return stripe.customers.createSource(customer, {source});
  }).then(response => {
      return event.data.adminRef.parent.set(response);
    }, error => {
      return event.data.adminRef.parent.child('error').set(userFacingMessage(error)).then(() => {
        return reportError(error, {user: user});
      });
  });
});

However I think they forgot to include this path n the delete method, as it's also expecting the string 'customer_id' value as the argument, according to the stripe API (https://stripe.com/docs/api#delete_customer).

// When a user deletes there account, clean up after them
exports.cleanupUser = functions.auth.user().onDelete(event => {
  return admin.database().ref(`/stripe_customers/${event.data.uid}`).once('value').then(snapshot => {
    return snapshot.val();
  }).then(customer => {
    return stripe.customers.del(customer);
  }).then(() => {
    return admin.database().ref(`/stripe_customers/${event.data.uid}`).remove();
  });
});

thanks

**Comment by SamMatthewsIsACommonName** [comment]
Here is the error from stackdriver using the original code:

```
Error: No such customer: [object Object]
at Error._Error (/user_code/node_modules/stripe/lib/Error.js:12)
at Error.Constructor (/user_code/node_modules/stripe/lib/utils.js:97)
at Error.Constructor (/user_code/node_modules/stripe/lib/utils.js:97)
at Function.StripeError.generate (Error.js:56)
at IncomingMessage.<anonymous> (/user_code/node_modules/stripe/lib/StripeResource.js:156)
at emitNone (events.js:91)
at IncomingMessage.emit (events.js:185)
at endReadableNT (_stream_readable.js:974)
at _combinedTickCallback (internal/process/next_tick.js:74)
at process._tickDomainCallback (next_tick.js:122)
```
As I said it is expecting a string ID argument not a whole customer object. Works fine now that I've changed.

**Comment by jamesdaniels** [comment]
Thanks for the report @SamMatthewsIsACommonName, I'll fix. We had a schema change and this was missed.

**Comment by SamMatthewsIsACommonName** [comment]
No worries, really appreciate the examples cheers! Also like I said in the database rules there is a small typo where you have the wild cards for sourceId and chargeId around the wrong way. Doesn't make any difference but might be confusing. Cheers

**Comment by cyrilzakka** [comment]
@SamMatthewsIsACommonName Thanks for that bit of code - was trying to debug things without realizing there was a log to Cloud Functions 😅

**Comment by jhuleatt** [comment]
Looks like this was fixed in #228 , closing
